edlib/src/edlib.cpp

```diff
@@ -1440,22 +1440,27 @@ static string transformSequences(const char* const queryOriginal, const int quer
         unsigned char c = static_cast<unsigned char>(queryOriginal[i]);
         if (!inAlphabet[c]) {
             inAlphabet[c] = true;
-            letterIdx[c] = static_cast<unsigned char>(alphabet.size());
-            alphabet += queryOriginal[i];
+            const unsigned char idx = static_cast<unsigned char>(alphabetSize++);
+            letterIdx[c] = idx;
+            alphabet[idx] = queryOriginal[i];
```

Comment:
I would if we could kick out `idx` and just use `alphabetSize` (and then ++ it at the very end).

Maybe if we made it unsigned char?

Reply:
If `alphabetSize` where an `unsigned char` then it would overflow to `0` if the alphabet needed `MAX_UCHAR` elements, in which case the final `std::string` would be passed a length of `0` which wouldn't be correct. Rather than using a specifically sized type I went with an `int` since it's guaranteed to be at least 16 bits and it's C/C++'s natural size for things like type promotion.

Reply:
Makes sense, thanks for explaining!